src/config.h

```diff
@@ -35,6 +37,46 @@ static inline const char *GCPolicy_ToString(GCPolicy policy) {
       return "huh?";  // LCOV_EXCL_LINE cannot be reached
   }
 }
+typedef struct {
+  size_t forkGcRunIntervalSec;
+  size_t forkGcCleanThreshold;
+  size_t forkGcRetryInterval;
+  size_t forkGcSleepBeforeExit;
+  int forkGCCleanNumericEmptyNodes;
+} forkGcConfig;
+
+typedef struct {
+  // If this is set, GC is enabled on all indexes (default: 1, disable with NOGC)
+  int enableGC;
+  size_t gcScanSize;
+  GCPolicy gcPolicy;
+
+  forkGcConfig forkGc;
+} GCConfig;
+
+// Configuration parameters related to aggregate request.
+typedef struct {
+  size_t maxSearchResults;
+  size_t maxAggregateResults;
+  // Default dialect level used throughout database lifetime.
+  unsigned int defaultDialectVersion;
+  // The maximal amount of time a single query can take before timing out, in milliseconds.
+  // 0 means unlimited
+  long long queryTimeoutMS;
+  RSTimeoutPolicy timeoutPolicy;
+} requestConfig;
+
+// Configuration parameters related to the query execution.
+typedef struct QueryConfig {
+  // The maximal number of expansions we allow for a prefix. Default: 200
+  long long maxPrefixExpansions;
+  // The minimal number of characters we allow expansion for in a prefix search. Default: 2
+  long long minTermPrefix;
+  long long maxResultsToUnsortedMode;
+  long long minUnionIterHeap;
+  // reply with time on profile
+  int printProfileClock;
+} QueryConfig;
```

Comment:
this is basically relevant to the iterators only? consider renaming to `iteratorsConfig`
I think it's a bit confusing to understand the difference between `queryConfig` and `requestConfig` ("request" is short for "query request" right?)